### PR TITLE
Filter Slashable Attester Public Keys in Slashing Interchange Import

### PR DESCRIPTION
**Pull request by rauljordan**
This is part of #7813. This PR filters public keys that are slashable with respect to other entries within the slashing interchange JSON file we are importing. It converts the attester's past attestations from the JSON file into their encoded, Prysm-specific form. Then, for each attestation, we check if it is slashable with respect to another one. If that's the case, we filter out the attester public keys that are slashable

### CODE REVIEW

**Comment by shayzluf** [comment]
@rauljordan 
from my understanding we should ignore slashable data on import.
validator may have signed a slashable offence that was not caught by any slasher. 
we want to protect it but not prevent its protection data import or validation. 
i thoink this test wont pass if we filter it out.
https://github.com/eth2-clients/slashing-protection-interchange-tests/blob/master/tests/generated/duplicate_pubkey_not_slashable.json

File: validator/slashing-protection/local/attesting-history/check_slashable_attestation.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
+package attestinghistory
```
Comment (rauljordan) [inline validator/slashing-protection/local/attesting-history/check_slashable_attestation.go:1]: Moved this into its own package due to import cycles in validator/client

File: validator/client/attest_protect.go [hunk 2]
```diff
@@ -151,142 +149,3 @@ func (v *validator) postAttSignUpdate(ctx context.Context, indexedAtt *ethpb.Ind
 	}
 	return v.db.SaveLowestSignedTargetEpoch(ctx, pubKey, indexedAtt.Data.Target.Epoch)
 }
-
-// isNewAttSlashable uses the attestation history to determine if an attestation of sourceEpoch
```
Comment (rauljordan) [inline validator/client/attest_protect.go:155]: Moved this into its own package due to cycles

File: validator/slashing-protection/local/standard-protection-format/import_test.go [hunk 3]
```diff
@@ -743,8 +745,163 @@ func Test_filterSlashablePubKeysFromBlocks(t *testing.T) {
...
+			}
+			for _, pk := range slashablePubKeys {
+				ok := wantedPubKeys[pk]
+				require.Equal(t, true, ok)
+			}
+		})
+	}
+}
+
+func Test_filterSlashablePubKeysFromAttestations(t *testing.T) {
```
Comment (rauljordan) [inline validator/slashing-protection/local/standard-protection-format/import_test.go:768]: Review these tests

File: validator/slashing-protection/local/standard-protection-format/import.go [hunk 4]
```diff
@@ -217,9 +223,43 @@ func filterSlashablePubKeysFromBlocks(ctx context.Context, blocksByPubkey map[[4
 	return slashablePubKeys
 }
 
-func filterSlashablePubKeysFromAttestations(ctx context.Context, attsByPubKey map[[48]byte][]*SignedAttestation) [][48]byte {
-	// TODO(#7813): Implement.
-	return nil
+func filterSlashablePubKeysFromAttestations(
```
Comment (rauljordan) [inline validator/slashing-protection/local/standard-protection-format/import.go:226]: Core logic of this PR is here

**Review by terencechain — APPROVED** [review]

**Pull request merged**